### DISCUSSION TITLE
AttributeError: module 'ecole' has no attribute 'instance'

### DISCUSSION THREAD

**Discussion by augustgao1995** [ds4dm/ecole#129] [Questions] [answered]
## Describe the bug

Dear professor,

Sorry for bothering.
I got this error, 
![截屏2021-02-08 下午7 10 45](https://user-images.githubusercontent.com/64518123/107193849-67b1f100-6a43-11eb-9a9d-9fba78799c1a.png)

But I have install ecole via this,
![截屏2021-02-08 下午7 11 30](https://user-images.githubusercontent.com/64518123/107194628-646b3500-6a44-11eb-9a5a-bca8375df5b2.png)

Could you help me? How do you think about this error? Thank you very much, professor. 

Yijie Gao (University of Sydney),
Sincerely.

## Setting
- OS: 
- Compiler: 
- Python version:3.8.2
- Ecole version:0.4.5

**Comment by AntoinePrv** [comment]
Hello @augustgao1995,

Nice to meet you. I'm personally a graduate student in Polytechnique Montréal.
You're not bothering at all, we want to make sure Ecole is bug-free.

I am not able to reproduce this problem with Ecole `0.4.5`, but this sounds like something that would happen with previous versions.
I'd first like to make sure of the version you are using. Can you run the following in your notebook:
```python
print("version: {v.major}.{v.minor}.{v.patch}@{v.git_revision}".format(v=ecole.core.get_build_version()))
```
On my computer, it prints
```
version: 0.4.5@1f7e82d73593e6d203d270378837c0996094dc40
```

**Comment by augustgao1995** [comment]
I got this.
![Q1](https://user-images.githubusercontent.com/64518123/107479272-ff901600-6bce-11eb-8128-422be3773dcd.jpeg)

by the way, I tried another install command of Ecole, and I got this.
![Q2](https://user-images.githubusercontent.com/64518123/107479335-15054000-6bcf-11eb-829c-b1eae79f39e4.jpeg)

**Comment by AntoinePrv** [comment]
The print command:
```python
print("version: {v.major}.{v.minor}.{v.patch}@{v.git_revision}".format(v=ecole.core.get_build_version()))
```
is Python code, so you'll need to run it in your Jupyter Notebook that you showed in your first message.

Otherwise, in you terminal, you can try the command:
```bash
python -m ecole.version
```

----

For the new install command you run, I am not sure what is going on, but one hypothesis I have is that we currently need to run the command with both `scipopt` and `conda-forge` channel at all time. Therefore you would need to do
```bash
conda install -c conda-forge -c scipopt ecole
```

If your conda environment is getting corrupted, perhaps you could create a new one?
```bash
conda create -n other_ecole -c conda-forge -c scipopt ecole pyscipopt
```

**Comment by augustgao1995** [comment]
Thank you for your patience and help. I got this error in Jupiter notebook.
![IMG_4352](https://user-images.githubusercontent.com/64518123/107746266-a4465b00-6d69-11eb-8571-7215ffc40db4.JPG)

**Comment by AntoinePrv** [comment, marked as answer]
In that last error message, `name 'ecole' is not defined`, Python is telling you that you are using some name (`ecole`) that is not in your scope.
Indeed, it does not exist yet in your notebook. You need to `import ecole` for that code to work.

---------

I cannot reproduce your error under any version of ecole since the instance generator were introduced (`v0.3.0`).
I think the best thing to do for you would be to try to reinstall everything from the start.
```bash
conda uninstall ecole
conda install -c conda-forge -c scipopt 'ecole>=0.4'
```

**Reply by augustgao1995** [reply]
I solve this, thank you very much.
